src/gluonts/ev/interface.py

```diff
@@ -0,0 +1,145 @@
+    ``gluonts.ev``.
+    """
+
+    forecast: Forecast
+
+    def __getitem__(self, name):
+        return np.expand_dims(self.forecast[name], axis=0)
+
+
+def _get_data_batch(
+    input_: DataEntry, label: DataEntry, forecast: Forecast
+) -> ChainMap:
+    forecast_dict = BatchForecast(forecast)
+
+    freq = forecast.start_date.freqstr
+    seasonality = get_seasonality(freq=freq)
+
+    other_data = {
+        "label": np.expand_dims(label["target"], axis=0),
+        "seasonal_error": np.expand_dims(
+            seasonal_error(input_["target"], seasonality=seasonality),
+            axis=0,
+        ),
+        "naive_2": np.expand_dims(
+            naive_2(input_["target"], len(label["target"]), freq=freq),
+            axis=0,
+        ),
+    }
+
+    return ChainMap(other_data, forecast_dict)
+
+
+def evaluate(
+    metrics, test_data: TestData, forecasts: Iterable[Forecast], *, axis=None
+) -> dict:
+    """
+    Evaluate ``metrics`` by comparing ``forecasts`` to ``test_data``.
+
+    The optional ``axis`` arguments controls aggregation of the metrics:
+    - ``None`` (default) aggregates across all dimensions
+    - ``0`` aggregates across the batch dimension, giving one metric value per forecast horizon
+    - ``1`` aggregates across the time dimension, giving one metric value per batch item
+
+    Return results as a dictionary.
+    """
+    assert axis is None or axis in [0, 1]
+
+    evaluators = {}
+    for metric in metrics:
+        evaluator = metric(axis=axis)
+        evaluators[evaluator.name] = evaluator
+
+    index_data = []
+    forecast_length = -1
+
+    for input_, label, forecast in zip(
+        test_data.input, test_data.label, forecasts
+    ):
+        forecast_length = len(forecast.index)
+
+        if axis == 1:
+            index_data.append((forecast.item_id, forecast.start_date))
+
+        data_batch = _get_data_batch(input_, label, forecast)
+
+        for evaluator in evaluators.values():
+            evaluator.update(data_batch)
+
+    metrics_dict = {
+        metric_name: evaluator.get()
+        for metric_name, evaluator in evaluators.items()
+    }
+
+    if axis is None:
+        metrics_dict["__index"] = {"name": None, "data": [None]}
+    elif axis == 0:
+        metrics_dict["__index"] = {
+            "name": "prediction horizon",
+            "data": list(range(forecast_length)),
+        }
+    elif axis == 1:
+        metrics_dict["__index"] = {
+            "name": ("item_id", "forecast time"),
+            "data": index_data,
+        }
+
+    return metrics_dict
+
+
+def evaluate_pd(
+    metrics, test_data: TestData, forecasts: Iterable[Forecast], *, axis=None
+) -> pd.DataFrame:
+    """
+    Call ``evaluate``, wrap result in a ``pd.DataFrame``.
+    """
+    metrics_dict = evaluate(
+        metrics=metrics,
+        test_data=test_data,
+        forecasts=forecasts,
+        axis=axis,
+    )
+
+    __index = metrics_dict.pop("__index")
```

Comment:
```suggestion
    index = metrics_dict.pop("__index")
```

Don't think we need to keep the `__` in a variable name.